packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,877 @@
+	}
+}
+
+function getTestTreeAsJSObject(
+	numberOfNodes: number,
+	shape: TreeShape,
+	dataType: TestPrimitives,
+): Jsonable {
+	let tree;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree = [getJSTestTreeDeep(numberOfNodes, dataType)];
+			break;
+		case TreeShape.Wide:
+			tree = getJSTestTreeWide(numberOfNodes, dataType);
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	return tree;
+}
+
+function getJSTestTreeWide(numberOfNodes: number, dataType: TestPrimitives): Jsonable {
+	const nodes = [];
+	for (let i = 0; i < numberOfNodes - 1; i++) {
+		const node = generateTreeData(dataType);
+		nodes.push(node);
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: nodes,
+		},
+		value: generateTreeData(dataType).value,
+	};
+	return tree;
+}
+
+function getJSTestTreeDeep(numberOfNodes: number, dataType: TestPrimitives): Jsonable {
+	const node = generateTreeData(dataType);
+	if (numberOfNodes === 1) {
+		return {
+			type: dataSchema.name,
+			value: node.value,
+		};
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: [getJSTestTreeDeep(numberOfNodes - 1, dataType)],
+		},
+		value: node.value,
+	};
+	return tree;
+}
+
+function readTreeAsJSObject(tree: Jsonable) {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) readTreeAsJSObject(tree[key]);
+		if (key === "value") {
+			assert(tree[key] !== undefined);
+		}
+	}
+}
+
+/**
+ * changes the value of the leaf node of the Jsonable tree.
+ * @param tree - tree in form of a Jsonable object
+ * @param shape - shape of the tree (wide vs deep)
+ */
+function manipulateTreeAsJSObject(tree: Jsonable, shape: TreeShape): void {
+	let nodesUnderRoot;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree[0].value = 123;
+			break;
+		case TreeShape.Wide:
+			nodesUnderRoot = tree.fields.foo.length;
+			if (nodesUnderRoot === 0) {
+				tree.fields.value = 123;
+			} else {
+				tree.fields.foo[nodesUnderRoot - 1].value = 123;
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+}
+
+function getLeafNodeFromJSObject(tree: Jsonable): Jsonable {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) {
+			if (tree[key].type !== undefined && tree[key].fields === undefined) {
+				return tree;
+			}
+			return getLeafNodeFromJSObject(tree[key]);
+		}
+	}
+}
+
+function readCursorTree(forest: IForestSubscription, numberOfNodes: number, shape: TreeShape) {
+	const readCursor = forest.allocateCursor();
+	moveToDetachedField(forest, readCursor);
+	assert(readCursor.firstNode());
+	switch (shape) {
+		case TreeShape.Deep:
+			for (let i = 0; i < numberOfNodes; i++) {
+				readCursor.firstField();
+				assert(readCursor.firstNode());
+			}
+			break;
+		case TreeShape.Wide:
+			if (numberOfNodes === 1) {
+				break;
+			}
+			readCursor.firstField();
+			readCursor.firstNode();
+			for (let j = 0; j < numberOfNodes; j++) {
+				readCursor.nextNode();
```

Comment:
you should check the returned value here. Might be better to write this in the: `for (let inNode = cursor.firstNode(); inNode; inNode = cursor.nextNode()) {` syle, then assert the loop ran the right number of times.

Reply:
changed for loop